### ISSUE TITLE
GetGeneric method causes crashes in BNM version 2.1.0

### ISSUE THREAD

**Issue by solomode0001** [ByNameModding/BNM-Android#71]
In BNM version 2.1.0, the application consistently crashes when using the GetGeneric method on classes or methods. This issue was not present in previous versions of the library.

log:
```
[SetupBNM] il2cpp::vm::Class::Init in lib: 0x1fb6a4.
[SetupBNM] code has il2cpp_image_get_class. BNM will use it.
[SetupBNM] il2cpp::vm::Class::FromIl2CppType in lib: 0x1fe16c.
[SetupBNM] il2cpp::vm::Type::GetClassOrElementClass in lib: 0x21cc58.
[SetupBNM] il2cpp::vm::Image::FromName in lib: 0x202660.
[SetupBNM] il2cpp::vm::Assembly::GetAllAssemblies in lib: 0x1fb888.
Class: [UnityEngine.CoreModule.dll]::[UnityEngine]::[Matrix3x3] - not found.
[CreateClass] Added new class (0xbea92300) [BNM.Coroutine]::[IEnumerator], parent - [System]::[Object], в [Assembly-CSharp.dll].
[CreateClass] Added new class (0xbea92400) [BNM.Coroutine]::[CustomWait], parent - [System]::[Object], в [Assembly-CSharp.dll].
hack_thread created
--------- beginning of crash
Invalid address 0xb6613f90 passed to free: value not allocated
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Version '2020.3.48f1 (b805b124c6b7)', Build type 'Release', Scripting Backend 'il2cpp', CPU 'armeabi-v7a'
Build fingerprint: 'Asus/Asus/ASUS_I003DD:9/mv-dev103041042:user/release-keys'
Revision: '0'
ABI: 'arm'
Timestamp: 2024-09-22 05:11:58+0700
pid: 5582, tid: 5748, name: UnityMain  >>> com.madfingergames.deadtrigger <<<
uid: 10102
signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
    r0  b6613f90  r1  00000001  r2  00000000  r3  0000ffff
    r4  0b52eefd  r5  00000001  r6  bea7f7a4  r7  00000014
    r8  b669ae60  r9  b6613fa4  r10 b6613f90  r11 bea7f798
    ip  0b52d210  sp  bea7f758  lr  0b4d90f4  pc  0b8a06c4

backtrace:
      #00 pc 000186c4  /system/lib/arm/nb/libc.so (free) (BuildId: f1478175a6e1d79b85bd0f9edf43ead1)
      #01 pc 000510f0  /data/app/com.madfingergames.deadtrigger-VZOkdpZOdx_FC6VXlSNMbQ==/lib/arm/libmain.so (BuildId: a0207e75e794bd65b6534e0c7a703a4333f2a80f)
      #02 pc 0005063c  /data/app/com.madfingergames.deadtrigger-VZOkdpZOdx_FC6VXlSNMbQ==/lib/arm/libmain.so (BuildId: a0207e75e794bd65b6534e0c7a703a4333f2a80f)
      #03 pc 00044144  /data/app/com.madfingergames.deadtrigger-VZOkdpZOdx_FC6VXlSNMbQ==/lib/arm/libmain.so (BuildId: a0207e75e794bd65b6534e0c7a703a4333f2a80f)
      #04 pc 00044be4  /data/app/com.madfingergames.deadtrigger-VZOkdpZOdx_FC6VXlSNMbQ==/lib/arm/libmain.so (BuildId: a0207e75e794bd65b6534e0c7a703a4333f2a80f)
      #05 pc 0007e65e  /system/lib/arm/nb/libtcb.so

other thread is trapped; signum = 6
FATAL EXCEPTION: UnityMain
Process: com.madfingergames.deadtrigger, PID: 5582
java.lang.Error: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Version '2020.3.48f1 (b805b124c6b7)', Build type 'Release', Scripting Backend 'il2cpp', CPU 'armeabi-v7a'
Build fingerprint: 'Asus/Asus/ASUS_I003DD:9/mv-dev103041042:user/release-keys'
Revision: '0'
ABI: 'arm'
Timestamp: 2024-09-22 05:11:58+0700
pid: 5582, tid: 5748, name: UnityMain  >>> com.madfingergames.deadtrigger <<<
uid: 10102
signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
    r0  b6613f90  r1  00000001  r2  00000000  r3  0000ffff
    r4  0b52eefd  r5  00000001  r6  bea7f7a4  r7  00000014
    r8  b669ae60  r9  b6613fa4  r10 b6613f90  r11 bea7f798
    ip  0b52d210  sp  bea7f758  lr  0b4d90f4  pc  0b8a06c4

backtrace:
      #00 pc 000186c4  /system/lib/arm/nb/libc.so (free) (BuildId: f1478175a6e1d79b85bd0f9edf43ead1)
      #01 pc 000510f0  /data/app/com.madfingergames.deadtrigger-VZOkdpZOdx_FC6VXlSNMbQ==/lib/arm/libmain.so (BuildId: a0207e75e794bd65b6534e0c7a703a4333f2a80f)
      #02 pc 0005063c  /data/app/com.madfingergames.deadtrigger-VZOkdpZOdx_FC6VXlSNMbQ==/lib/arm/libmain.so (BuildId: a0207e75e794bd65b6534e0c7a703a4333f2a80f)
      #03 pc 00044144  /data/app/com.madfingergames.deadtrigger-VZOkdpZOdx_FC6VXlSNMbQ==/lib/arm/libmain.so (BuildId: a0207e75e794bd65b6534e0c7a703a4333f2a80f)
      #04 pc 00044be4  /data/app/com.madfingergames.deadtrigger-VZOkdpZOdx_FC6VXlSNMbQ==/lib/arm/libmain.so (BuildId: a0207e75e794bd65b6534e0c7a703a4333f2a80f)
      #05 pc 0007e65e  /system/lib/arm/nb/libtcb.so

	at libc.free(free:0)
	at libmain.0x510f0(Native Method)
	at libmain.0x5063c(Native Method)
	at libmain.0x44144(Native Method)
	at libmain.0x44be4(Native Method)
	at libtcb.0x7e65e(Native Method)
```

code:
```c++
void hack_thread() {
    LOGI(OBFUSCATE("hack_thread created"));

    BNM::Class gameObject("UnityEngine", "GameObject");
    auto getComponent = gameObject.GetMethod("GetComponent", 0).GetGeneric({BNM::CompileTimeClassBuilder(NULL, "ComponentPlayer").Build()});

    BNM_LOG_INFO("offset = %p", BNM::Utils::OffsetInLib((void*)getComponent.GetOffset()));
}
```

**Comment by BNM-Dev** [comment]
In v2.1.1 should be fixed.

**Comment by solomode0001** [comment]
still crash sir

log:
```
2024-09-22 15:41:54.143  5957-6017  ByNameModding           com.madfingergames.deadtrigger       D  [SetupBNM] il2cpp::vm::Class::Init in lib: 0x1fb6a4.
2024-09-22 15:41:54.143  5957-6017  ByNameModding           com.madfingergames.deadtrigger       D  [SetupBNM] code has il2cpp_image_get_class. BNM will use it.
2024-09-22 15:41:54.144  5957-6017  ByNameModding           com.madfingergames.deadtrigger       D  [SetupBNM] il2cpp::vm::Class::FromIl2CppType in lib: 0x1fe16c.
2024-09-22 15:41:54.144  5957-6017  ByNameModding           com.madfingergames.deadtrigger       D  [SetupBNM] il2cpp::vm::Type::GetClassOrElementClass in lib: 0x21cc58.
2024-09-22 15:41:54.144  5957-6017  ByNameModding           com.madfingergames.deadtrigger       D  [SetupBNM] il2cpp::vm::Image::FromName in lib: 0x202660.
2024-09-22 15:41:54.144  5957-6017  ByNameModding           com.madfingergames.deadtrigger       D  [SetupBNM] il2cpp::vm::Assembly::GetAllAssemblies in lib: 0x1fb888.
2024-09-22 15:41:54.157  5957-6017  ByNameModding           com.madfingergames.deadtrigger       W  Class: [UnityEngine.CoreModule.dll]::[UnityEngine]::[Matrix3x3] - not found.
2024-09-22 15:41:54.160  5957-6017  ByNameModding           com.madfingergames.deadtrigger       I  [CreateClass] Added new class (0xbe7c3200) [BNM.Coroutine]::[IEnumerator], parent - [System]::[Object], в [Assembly-CSharp.dll].
2024-09-22 15:41:54.160  5957-6017  ByNameModding           com.madfingergames.deadtrigger       I  [CreateClass] Added new class (0xbe7c3300) [BNM.Coroutine]::[CustomWait], parent - [System]::[Object], в [Assembly-CSharp.dll].
2024-09-22 15:41:54.160  5957-6017  ByNameModding           com.madfingergames.deadtrigger       I  hack_thread created
2024-09-22 15:41:54.161  5957-6017  ByNameModding           com.madfingergames.deadtrigger       I  GameObject::GetComponent<T>: 0x0
2024-09-22 15:41:54.161  5957-6017  ByNameModding           com.madfingergames.deadtrigger       I  GameObject::GetComponent<object>: 0x0
2024-09-22 15:41:54.161  5957-6017  ByNameModding           com.madfingergames.deadtrigger       I  Dictionary<TKey, TValue>::Add: 0x0
--------- beginning of crash
2024-09-22 15:41:54.161  5957-6017  ByNameModding           com.madfingergames.deadtrigger       I  Dictionary<int, int>::Add: 0x0
2024-09-22 15:41:54.161  5957-6017  libc                    com.madfingergames.deadtrigger       A  stack corruption detected (-fstack-protector)
2024-09-22 15:41:54.163  5957-6017  CRASH                   com.madfingergames.deadtrigger       E  *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
2024-09-22 15:41:54.163  5957-6017  CRASH                   com.madfingergames.deadtrigger       E  Version '2020.3.48f1 (b805b124c6b7)', Build type 'Release', Scripting Backend 'il2cpp', CPU 'armeabi-v7a'
2024-09-22 15:41:54.164  5957-6017  CRASH                   com.madfingergames.deadtrigger       E  Build fingerprint: 'Asus/Asus/ASUS_I003DD:9/mv-dev103041042:user/release-keys'
2024-09-22 15:41:54.164  5957-6017  CRASH                   com.madfingergames.deadtrigger       E  Revision: '0'
2024-09-22 15:41:54.164  5957-6017  CRASH                   com.madfingergames.deadtrigger       E  ABI: 'arm'
2024-09-22 15:41:54.166  5957-6017  CRASH                   com.madfingergames.deadtrigger       E  Timestamp: 2024-09-22 15:41:54+0700
2024-09-22 15:41:54.167  5957-6017  CRASH                   com.madfingergames.deadtrigger       E  pid: 5957, tid: 6017, name: UnityMain  >>> com.madfingergames.deadtrigger <<<
2024-09-22 15:41:54.167  5957-6017  CRASH                   com.madfingergames.deadtrigger       E  uid: 10102
2024-09-22 15:41:54.167  5957-6017  CRASH                   com.madfingergames.deadtrigger       E  signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
2024-09-22 15:41:54.169  5957-6017  CRASH                   com.madfingergames.deadtrigger       E      r0  9137e9b8  r1  c6191000  r2  e5e98f60  r3  00000000
2024-09-22 15:41:54.169  5957-6017  CRASH                   com.madfingergames.deadtrigger       E      r4  c617e5ec  r5  0425af48  r6  c3c136d0  r7  05d4c6d0
2024-09-22 15:41:54.169  5957-6017  CRASH                   com.madfingergames.deadtrigger       E      r8  00000001  r9  c6196200  r10 c617efa0  r11 c617ef4c
2024-09-22 15:41:54.169  5957-6017  CRASH                   com.madfingergames.deadtrigger       E      ip  05d16518  sp  c617e5e8  lr  05281e2f  pc  0b6f6a38
2024-09-22 15:41:54.170  5957-6017  CRASH                   com.madfingergames.deadtrigger       E  
                                                                                                    backtrace:
2024-09-22 15:41:54.171  5957-6017  CRASH                   com.madfingergames.deadtrigger       E        #00 pc 00072a38  /system/lib/arm/nb/libc.so (__stack_chk_fail+4) (BuildId: f1478175a6e1d79b85bd0f9edf43ead1)
2024-09-22 15:41:54.421  5957-6017  CRASH                   com.madfingergames.deadtrigger       E  Tombstone written to: /storage/emulated/0/Android/data/com.madfingergames.deadtrigger/files/tombstone_02
2024-09-22 15:41:54.421  5957-6017  AndroidRuntime          com.madfingergames.deadtrigger       E  FATAL EXCEPTION: UnityMain
                                                                                                    Process: com.madfingergames.deadtrigger, PID: 5957
                                                                                                    java.lang.Error: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
                                                                                                    Version '2020.3.48f1 (b805b124c6b7)', Build type 'Release', Scripting Backend 'il2cpp', CPU 'armeabi-v7a'
                                                                                                    Build fingerprint: 'Asus/Asus/ASUS_I003DD:9/mv-dev103041042:user/release-keys'
                                                                                                    Revision: '0'
                                                                                                    ABI: 'arm'
                                                                                                    Timestamp: 2024-09-22 15:41:54+0700
                                                                                                    pid: 5957, tid: 6017, name: UnityMain  >>> com.madfingergames.deadtrigger <<<
                                                                                                    uid: 10102
                                                                                                    signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
                                                                                                        r0  9137e9b8  r1  c6191000  r2  e5e98f60  r3  00000000
                                                                                                        r4  c617e5ec  r5  0425af48  r6  c3c136d0  r7  05d4c6d0
                                                                                                        r8  00000001  r9  c6196200  r10 c617efa0  r11 c617ef4c
                                                                                                        ip  05d16518  sp  c617e5e8  lr  05281e2f  pc  0b6f6a38
                                                                                                    
                                                                                                    backtrace:
                                                                                                          #00 pc 00072a38  /system/lib/arm/nb/libc.so (__stack_chk_fail+4) (BuildId: f1478175a6e1d79b85bd0f9edf43ead1)
                                                                                                    
                                                                                                    	at libc.__stack_chk_fail(__stack_chk_fail:4)
```

code:
```c++

void OnLoaded_Example_03() {
    using namespace BNM;

    // ---------------------------------------------------- Generic methods ----------------------------------------------------

    auto object = BNM::Defaults::Get<void *>().ToClass();

    auto gameObject = Class(OBFUSCATE_BNM("UnityEngine"), OBFUSCATE_BNM("GameObject"), Image(OBFUSCATE_BNM("UnityEngine.CoreModule")));

    //! There is GameObject class with the generic GetComponent method:
    /*
      public class GameObject : Object {
        // ...
		public T GetComponent<T>() {}
        // ...
	  }
    */

    // So we get `public T GetComponent<T>() {}`
    auto GetComponent = gameObject.GetMethod(OBFUSCATE_BNM("GetComponent"), 0);

    // And so - `public object GetComponent<object>(){}`
    Method<void *> GetComponentObject = GetComponent.GetGeneric({object});

    // Let's check results by getting address of method
    BNM_LOG_INFO("GameObject::GetComponent<T>: 0x%lX", (BNM_PTR)BNM::Utils::OffsetInLib((void *)GetComponent.GetOffset()));
    BNM_LOG_INFO("GameObject::GetComponent<object>: 0x%lX", (BNM_PTR)BNM::Utils::OffsetInLib((void *)GetComponentObject.GetOffset()));


    // ---------------------------------------------------- Generic классы (classes) ----------------------------------------------------

    //! Let's analyze getting of generic classes in the code from example 02:
    /*
    namespace System.Collections.Generic
    {
      public class Dictionary<TKey, TValue> : ... {}
    }
    */


    // To find a type with <T1, T2, T3, ..., Tx>, you need to search for 'TypeName`(number of parameters in <>)"

    // So we get `Dictionary<TKey, TValue>`. Since the Dictionary class has two genetic types, then after '`' we write '2'
    auto dictionaryClass = BNM::Class(OBFUSCATE_BNM("System.Collections.Generic"), OBFUSCATE_BNM("Dictionary`2"), BNM::Image(OBFUSCATE_BNM("mscorlib.dll")));

    // And so - `Dictionary<int, int>`
    auto dictionary_int_int_Class = dictionaryClass.GetGeneric({BNM::Defaults::Get<int>(), BNM::Defaults::Get<int>()});

    // Let's check the results using the address of the Add method as an example
    BNM_LOG_INFO("Dictionary<TKey, TValue>::Add: 0x%lX", (BNM_PTR)BNM::Utils::OffsetInLib((void *)dictionaryClass.GetMethod("Add", 2).GetOffset()));
    BNM_LOG_INFO("Dictionary<int, int>::Add: 0x%lX", (BNM_PTR)BNM::Utils::OffsetInLib((void *)dictionary_int_int_Class.GetMethod("Add", 2).GetOffset()));
}

void hack_thread() {
    BNM_LOG_INFO(OBFUSCATE("hack_thread created"));

    OnLoaded_Example_03();
}
```

**Comment by yunli97** [comment]
You can try changing it to something like this
![image](https://github.com/user-attachments/assets/369db55d-1a4a-423a-aa82-8b49f7bceb74)

**Comment by BNM-Dev** [comment]
Fixed in v2.1.2. Tested in test Unity 2020.3.48f1 project.
